lib/platform/azure/index.ts

```diff
@@ -267,20 +271,43 @@ export async function getBranchPr(branchName: string): Promise<Pr | null> {
   return existingPr ? getPr(existingPr.number) : null;
 }
 
-export async function getBranchStatusCheck(
-  branchName: string,
-  context: string
-): Promise<BranchStatus> {
-  logger.trace(`getBranchStatusCheck(${branchName}, ${context})`);
+async function getStatusCheck(branchName: string): Promise<GitStatus[]> {
   const azureApiGit = await azureApi.gitApi();
   const branch = await azureApiGit.getBranch(
     config.repoId,
     getBranchNameWithoutRefsheadsPrefix(branchName)
   );
-  if (branch.aheadCount === 0) {
-    return BranchStatus.green;
+  // only grab the latest statuses, it will group any by context
+  return await azureApiGit.getStatuses(
```

Comment:
I made this suggestion earlier, and it broke an eslint rule :facepalm: 

```suggestion
  return azureApiGit.getStatuses(
```